renku/core/models/projects.py

```diff
@@ -88,8 +98,52 @@ def __attrs_post_init__(self):
                 # this assumes the project is being newly created
                 self.creator = Creator.from_git(self.client.repo)
 
-        if not self._id and self.client:
-            self._id = self.client.project_id
+        if not self._id or self._id.startswith('file://'):
+            self._id = self.project_id
+
+    @property
+    def project_id(self):
+        """Return the id for the project based on the repo origin remote."""
+        from renku.core.models.git import GitURL
+
+        # RENKU_DOMAIN environment variable is the host override
+        host = os.environ.get('RENKU_DOMAIN')
+        owner = self.creator.short_name if self.creator else 'NULL'
+        name = self.name
+
+        if self.client and not host:
+            remote_name = 'origin'
+            try:
+                remote_branch = \
+                    self.client.repo.head.reference.tracking_branch()
+                if remote_branch is not None:
+                    remote_name = remote_branch.remote_name
+            except TypeError:
+                pass
+
+            try:
+                url = GitURL.parse(self.client.repo.remotes[remote_name].url)
+
+                # Remove gitlab. unless running on gitlab.com.
+                hostname_parts = url.hostname.split('.')
+                if len(hostname_parts) > 2 and hostname_parts[0] == 'gitlab':
+                    hostname_parts = hostname_parts[1:]
+                url = attr.evolve(url, hostname='.'.join(hostname_parts))
+            except IndexError:
+                url = None
+
+            if url:
+                # set the host if not set from RENKU_DOMAIN above
+                host = host or url.hostname or 'localhost'
+                owner = url.owner or owner
+                name = url.name or name
+
+        return 'https://{host}/{project_url_path}/{owner}/{name}'.format(
```

Comment:
Can we build URL's in the same way as we do in [here](https://github.com/SwissDataScienceCenter/renku-python/blob/master/renku/core/commands/providers/zenodo.py#L49).

Reply:
done